post-processing-code/general-functions/aggregation.py

```diff
@@ -0,0 +1,167 @@
+import glob
+import csv
+import pandas as pd
+import numpy as np
+
+AGGEGATE_DEFAULT_TYPING_MAP = {"year_id": float, "age_start": float, "age_end": float,
+                               "mean": float, "2.5_percentile": float, "5_percentile": float,
+                               "10_percentile": float, "25_percentile": float, "50_percentile": float,
+                               "75_percentile": float, "90_percentile": float, "95_percentile": float,
+                               "97.5_percentile": float, "median": float, "standard_deviation": float
+                               }
+
+def percentile(n):
+        def percentile_(x):
+            return np.percentile(x, n)
+        percentile_.__name__ = '%s_percentile' % n
+        return percentile_
+def calc_not_na(x):
+    return x.notnull().mean() * 100
+
+def aggregatePostProcessedFiles(
+    pathToOutputFiles: str = ".",
+    specific_files: str = "*.csv",
+    columns_to_replace_with_nan: list[str] = ["mean"],
```

Comment:
Its not necessarily deleting columns, just properly typing them to a given format (i.e NaN vs None). However, I moved this to the the iu_lvl_aggregate function, so that its no longer a no-op. In essence the iu-lvl-aggregate is now a wrapper function for this.